lib/python/script/utils.py

```diff
@@ -467,3 +471,104 @@ def clock():
     if sys.version_info > (3,2):
         return time.perf_counter()
     return time.clock()
+
+
+def legalize_vector_name(name):
+    if not name:
+        raise ValueError("name cannot be empty")
+    if re.match("[^A-Za-z0-9]", name[0]):
+        name[0] = "_"
+    re.sub("[^A-Za-z0-9_]", "_", name)
+    keywords = ["and", "or", "not"]
+    if name in keywords:
+        name = "{name}_".format(**locals())
```

Comment:
Why not just: `name = "{}_".format(name)`?

Reply:
`"{}_".format(name)` is fine with me, but all other places there use `.format(**locals())` so I kept it there. In the future, these can be nicely converted to f-strings.